crux-metrics/src/crux/metrics.clj

```diff
@@ -18,28 +20,57 @@
+                               (jmx/start-reporter registry args))
+                   :args {:crux.metrics.dropwizard.jmx/domain {:doc "Add custom domain"
+                                                               :required? false
+                                                               :crux.config/type :crux.config/string-vector}
+                          :crux.metrics.dropwizard.jmx/rate-unit {:doc "Set rate unit"
+                                                                  :required? false
+                                                                  :crux.config/type :crux.config/string}
+                          :crux.metrics.dropwizard.jmx/duration-unit {:doc "Set duration unit"
+                                                                      :required? false
+                                                                      :crux.config/type :crux.config/string}}
                    :deps #{::registry}}})
 
 (def console-reporter
-  {::console-reporter {:start-fn (fn [{::keys [registry]} {::keys [console-reporter-opts console-reporter-rate]}]
-                                   (doto (console/reporter registry (merge {} console-reporter-opts))
-                                     (console/start (or console-reporter-rate 1))))
+  {::console-reporter {:start-fn (fn [{::keys [registry]} args]
+                                   (console/start-reporter registry args))
                        :deps #{::registry}}})
 
 (def csv-reporter
-  {::csv-reporter {:start-fn (fn [{::keys [registry]} {::keys [csv-reporter-opts csv-reporter-file csv-reporter-rate]}]
-                               (doto (csv/reporter registry
-                                                   (or csv-reporter-file "/tmp/csv_reporter")
-                                                   (merge {}
-                                                          csv-reporter-opts))
-                                 (csv/start (or csv-reporter-rate 1))))
+  {::csv-reporter {:start-fn (fn [{::keys [registry]} args]
+                               (csv/start-reporter registry args))
                    :deps #{::registry}}})
 
+;; TODO a decision for consistency, should args be a flat map in the config, or
+;; contained in a arg themself
+
+(def cloudwatch-reporter
+  {::cloudwatch-reporter {:start-fn (fn [{::keys [registry]}
+                                         {:crux.metrics.dropwizard.cloudwatch/keys
+                                          [duration]
+                                          :as args}]
+                                      (let [cw-rep (cloudwatch/report registry args)]
+                                        (cloudwatch/start cw-rep (Duration/parse duration))))
+                          :deps #{::registry}
+                          :args {:crux.metrics.dropwizard.cloudwatch/region {:doc "Region for uploading metrics. Tries to get it using api. If this fails, you will need to specify region."
```

Comment:
let's alias these namespaces so we don't have to write them out every time? :smile: